sql/tables.go

```diff
@@ -98,7 +98,7 @@ type IndexAddressable interface {
 	// IndexedAccess returns a table that can perform scans constrained to
 	// an IndexLookup on the index given, or nil if the index cannot support
 	// the lookup expression.
-	IndexedAccess(IndexLookup) IndexedTable
+	IndexedAccess(ctx *Context, lookup IndexLookup) (IndexedTable, error)
```

Comment:
this is the only change i don't like. to get to this point 1) the session-specific table has told us it has and supports this index, 2) the index accepts a very specific key expression. if this fails, either 1 or 2 should have failed loudly

Reply:
I'll see if I can get rid of it with later changes. It's only needed by the memory table implementation, since it needs to get the edit accumulator from the session so it can apply those changes before it can returned an indexed version.

Reply:
Took most of the afternoon but I found a way around this. Reverted to original.